### PR TITLE
:bug: Remove defaults from crd v1beta1

### PR DESCRIPTION
**Pull request by tamalsaha**
Fixes https://github.com/kubernetes-sigs/controller-tools/issues/445

Signed-off-by: Tamal Saha <tamal@appscode.com>

### CODE REVIEW

**Comment by vincepri** [comment]
It seems a little strange to me that we'd silently remove the defaults markers for an unsupported type, this could lead into confusion, thoughts @alvaroaleman @DirectXMan12 ?

File: pkg/crd/gen.go [hunk 1]
```diff
@@ -182,6 +189,59 @@ func toTrivialVersions(crd *apiextlegacy.CustomResourceDefinition) {
 	crd.Spec.AdditionalPrinterColumns = canonicalColumns
 }
 
+// removeDefaults removes defaults from apiextensions.k8s.io/v1beta1 CRD definition.
+// To use defaulting, CustomResourceDefinitions must use API version apiextensions.k8s.io/v1
+func removeDefaults(schema *apiextlegacy.JSONSchemaProps) {
```
Comment (DirectXMan12) [inline pkg/crd/gen.go:194]: we've got a schema visitor for this purpose, so you shouldn't have to manually write a schema visitor.

**Review by DirectXMan12 — CHANGES_REQUESTED** [review]
left comment, needs a test case, also we should probably warn about this, or maybe just error out.
I'd lean towards just erroring out, but I wouldn't be surprised if people have workarounds (deploying different controller versions or webhooks but same YAML), so a warning is probably better.

**Comment by DirectXMan12** [comment]
Closed by #481, but that'll need some fixups.

**Pull request closed without merging**